rust/connlib/tunnel/benches/tunnel.rs

```diff
@@ -0,0 +1,180 @@
+        tokio::fs::create_dir_all(wintun_path.parent().unwrap())
+            .await
+            .unwrap();
+        tokio::fs::write(&wintun_path, connlib_shared::windows::wintun_bytes())
+            .await
+            .unwrap();
+
+        let mut tun = Tun::new()?;
+
+        const MTU: usize = 1_280;
+        const NUM_REQUESTS: u64 = 1_000;
+        const REQ_CODE: u8 = 42;
+        const REQ_LEN: usize = 1_000;
+        const RESP_CODE: u8 = 43;
+        const SERVER_PORT: u16 = 3000;
+
+        let ipv4 = Ipv4Addr::from([100, 90, 215, 97]);
+        let ipv6 = Ipv6Addr::from([0xfd00, 0x2021, 0x1111, 0x0, 0x0, 0x0, 0x0016, 0x588f]);
+        let mut device_manager =
+            connlib_shared::tun_device_manager::platform::TunDeviceManager::new()?;
+        device_manager.set_ips(ipv4, ipv6).await?;
+        tun.add_route(ipv4.into())?;
+
+        let server_addr = (ipv4, SERVER_PORT).into();
+
+        // Listen for incoming packets on Wintun, and echo them.
+        let server_task = tokio::spawn(async move {
+            tracing::debug!("Server task entered");
+            let mut requests_served = 0;
+            // We aren't interested in allocator speed or doing any processing,
+            // so just cache the response packet
+            let mut response_pkt = None;
+            let mut time_spent = Duration::from_millis(0);
+            loop {
+                let mut req_buf = [0u8; MTU];
+                poll_fn(|cx| tun.poll_read(&mut req_buf, cx)).await?;
+                let start = Instant::now();
+                let original_pkt = IpPacket::new(&req_buf).unwrap();
+                let Some(original_udp) = original_pkt.as_udp() else {
+                    continue;
+                };
+                if original_udp.get_destination() != SERVER_PORT {
+                    continue;
+                }
+                if original_udp.payload()[0] != REQ_CODE {
+                    panic!("Wrong request code");
+                }
+
+                // Only generate the response packet on the first loop,
+                // then just reuse it.
+                let res_buf = response_pkt
+                    .get_or_insert_with(|| {
+                        ip_packet::make::udp_packet(
+                            original_pkt.destination(),
+                            original_pkt.source(),
+                            original_udp.get_destination(),
+                            original_udp.get_source(),
+                            vec![RESP_CODE],
+                        )
+                    })
+                    .packet();
+                tun.write4(res_buf)?;
+                requests_served += 1;
+                time_spent += start.elapsed();
+                if requests_served >= NUM_REQUESTS {
+                    break;
+                }
+            }
+
+            tracing::info!(time_spent = format!("{:?}", time_spent), "Server all good");
```

Comment:
```suggestion
            tracing::info!(?time_spent, "Server all good");
```

Isn't this equivalent?

Reply:
Yeah, for some reason I thought I couldn't get `Duration` to log correctly. Sometimes the compiler says it doesn't implement `Value` and sometimes it works.